src/expand.ts

```diff
@@ -79,9 +91,33 @@ export interface ExpansionOptions {
  * @returns A string with the variable references replaced
  */
 export async function expandString(tmpl: string, opts: ExpansionOptions) {
+  if (!tmpl) {
+    return tmpl;
+  }
+
+  const MAX_RECURSION = 50;
```

Comment:
This seems a bit high for variable expansion.  A smaller number like 5 or 10 seems more reasonable.

Reply:
Yeah randomly picked 50... 10 sounds good to me, and we can easily adjust this based on the customers' feedbacks.